scripts/core/CONTRIB.rst

```diff
@@ -219,13 +219,38 @@ implement your experimental feature.
 Naming Convention
 -----------------
 
+The following naming conventions must be followed:
+
+## --validate=off
+    *   All functions must be prefixed with ``${x}``
+    *   All functions must use camel case ``${x}ObjectAction`` convention
+    *   All macros must use all caps ``${X}_NAME`` convention
+    *   All structures, enumerations and other types must follow ``${x}_name_t`` 
+        snake case convention
+    *   All structure members and function parameters must use camel case 
+        convention
+    *   All enumerator values must use all caps ``${X}_ENUM_ETOR_NAME`` 
+        convention
+    *   All handle types must end with ``handle_t``
+    *   All descriptor structures must end with ``desc_t``
+    *   All property structures must end with ``properties_t``
+    *   All flag enumerations must end with ``flags_t``
+## --validate=on
+
+The following coding conventions must be followed:
+
+    *   All descriptor structures must be derived from ${x}_base_desc_t
+    *   All property structures must be derived from ${x}_base_properties_t
+    *   All function input parameters must precede output parameters
+    *   All functions must return ${x}_result_t
+
 In addition to the requirements referred to in the `Writing YAML`_ section, and
 to easily differentiate experimental feature symbols, the following conventions
 *must* be adhered to when defining experimental features:
 
 ## --validate=off
-*   All functions must use camel case ``${x}ObjectActionExp`` convention.
-*   All macros must use all caps ``${X}_NAME_EXP`` convention.
-*   All structures, enumerations, and other types must follow
-    ``${x}_exp_name_t`` name case convention.
+    *   All functions must use camel case ``${x}ObjectActionExp`` convention.
+    *   All macros must use all caps ``${X}_NAME_EXP`` convention.
+    *   All structures, enumerations, and other types must follow
+        ``${x}_exp_name_t`` name case convention.
```

Comment:
I deliberately did not indent these bullet points to become a block quote, I think it looks bad in the rendered output.

Reply:
Fixed.